### PR TITLE
Adding new version 2020-07-01-preview for DataMove

### PR DESCRIPTION
**Pull request by pratimaupadhyay02**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Service team MUST add the "**WaitForARMFeedback**" label if the management plane API changes fall into one of the below categories. 
  - adding/removing APIs.
  - adding/removing properties.
  - adding/removing API-version. 
  - adding a new service in Azure.

<i>Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.</i>

- [x] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Review by majastrz — COMMENTED** [review]
Please go through the checklist in the PR description and ensure that everything is passing. Won't be able to review from ARM side until then.

**Comment by isra-fel** [comment]
There are 2 `ProtectionPolicies_Get` operations. The one in 2019-06-15 is changed to long running operation in this PR, the other in [2019-05-13](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2019-05-13/bms.json#L649) is still synchronous. The fact that these two versions were generated together caused .NET SDK to fail I suppose.

What's the reason to change this operation to LRO? Is it possible to change 2019-05-13 as well?

**Comment by chiragg4u** [comment]
Please go through the contribution checklist added as part of the review and mark them completed. This is required for review process.

**Comment by chiragg4u** [comment]
For setting up new version we recommend to start with a commit of the previous version, to allow easy review.

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2019-06-15/bms.json [hunk 1]
```diff
@@ -746,7 +746,7 @@
         "tags": [
           "ProtectionPolicies"
         ],
-        "description": "Creates or modifies a backup policy. This is an asynchronous operation. Status of the operation can be fetched\r\nusing GetPolicyOperationResult API.",
```
Comment (sumitmal2711) [inline specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2019-06-15/bms.json:749]: Can you undo these changes? We can undo changes that are not part of DS Move work

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2020-07-01/bms.json [hunk 2]
```diff
@@ -0,0 +1,723 @@
...
+          "description": "Error code of the operation failure.",
+          "type": "string"
+        },
+        "message": {
+          "description": "Error message displayed if the operation failure.",
+          "type": "string"
+        }
+      }
+    },
+    "OperationStatusExtendedInfo": {
```
Comment (sumitmal2711) [inline specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2020-07-01/bms.json:392]: QQ: Why do we need these in new file? Dont they exist in existing files.

Comment (pratimaupadhyay02) [reply specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2020-07-01/bms.json:392]: > QQ: Why do we need these in new file? Dont they exist in existing files.

These are referenced in GetOperationStatus API for data move. We can change that reference to the older versions if we remove them from this file. As long as the definition in the older file and this file is same, it is not causing any issue.

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2019-06-15/bms.json (hunk 1 above)
Comment (pratimaupadhyay02) [reply specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2019-06-15/bms.json]: Done

> Can you undo these changes? We can undo changes that are not part of DS Move work

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2020-07-01/bms.json [hunk 3]
```diff
@@ -0,0 +1,479 @@
...
+          }
+        },
+        "x-ms-examples": {
+          "Get OperationStatus": {
+            "$ref": "./examples/BackupDataMove/BackupDataMoveOperationStatus_Get.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{vaultName}/backupDataMove/prepare": {
```
Comment (chiragg4u) [inline specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2020-07-01/bms.json:72]: >backupDataMove [](start = 128, length = 14)

As discussed in the meeting, this looks like we are introducing a type backupDataMove but there is no object being created. As we discussed you can possibly model it as dataMoveJobs object which can be PUT/GET and possibly DELETED 

Alternatively you can provide these actions directly to the {vault} itself something like 
vaults/{vaultName}/prepareMove
vaults/{vaultName}/triggerMove

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2020-07-01/bms.json [hunk 4]
```diff
@@ -0,0 +1,479 @@
...
+      "post": {
+        "tags": [
+          "DataMove"
+        ],
+        "description": "Prepare Data Move controller",
```
Comment (chiragg4u) [inline specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2020-07-01/bms.json:77]: Add more detail about this needs to be called on the source vault etc. Also as discussed add more comments as it will be used to generate documentation.

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2020-07-01/bms.json [hunk 5]
```diff
@@ -0,0 +1,479 @@
...
+            "Vault",
+            "Container"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "DataMoveLevel",
+            "modelAsString": true
+          }
+        },
+        "sourceContainerArmIds": {
```
Comment (chiragg4u) [inline specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2020-07-01/bms.json:310]: Please identify required property.

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2020-07-01/bms.json [hunk 6]
```diff
@@ -0,0 +1,479 @@
...
+          "description": "Ignore the artifacts which are already moved.",
+          "type": "boolean"
+        }
+      }
+    },
+    "PrepareDataMoveRequestResource": {
+      "description": "Prepare DataMove Request",
+      "allOf": [
+        {
+          "$ref": "../2019-06-15/bms.json#/definitions/Resource"
```
Comment (chiragg4u) [inline specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2020-07-01/bms.json:327]: >    "$ref": "../2019-06-15/bms.json#/definitions/Resource" [](start = 6, length = 58)

Is this a full resource? Unless you are modeling as a resource you don't need id and other attributes for  this request object.

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2020-07-01/bms.json [hunk 7]
```diff
@@ -0,0 +1,479 @@
...
+      "properties": {
+        "targetResourceId": {
+          "description": "ARM Id of target vault",
+          "type": "string"
+        },
+        "targetRegion": {
+          "description": "Target Region",
+          "type": "string"
+        },
+        "targetCloud": {
```
Comment (chiragg4u) [inline specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2020-07-01/bms.json:293]: >targetCloud [](start = 9, length = 11)

Since you won't be implementing this functionality anytime soon, I would recommend removing it. This can be added later easily. IMO this would be a bigger feature and would require other updates also.

**Comment by chiragg4u** [comment]
Changes looks fine from ARM API perspective. Signing off.

**Review by chiragg4u — APPROVED** [review]

**Review by lirenhe — APPROVED** [review]
